lib/data/external/GCP/GcpApis/copyObject.js

```diff
@@ -0,0 +1,112 @@
+const async = require('async');
+const request = require('request');
+const { errors } = require('arsenal');
+
+const { logger, JsonError, checkSourceString } = require('../GcpUtils');
+const { logHelper } = require('../../utils');
+
+/**
+ * copyObject - minimum required functionality to perform object copy
+ * for GCP Backend
+ * @param {object} params - update metadata params
+ * @param {string} params.Bucket - bucket name
+ * @param {string} params.Key - object key
+ * @param {string} param.CopySource - source object
+ * @param {function} callback - callback function to call with the copy object
+ * result
+ * @return {undefined}
+ */
+function _copyObject(params, callback) {
+    const { CopySource } = params;
+    if (!checkSourceString(CopySource)) {
```

Comment:
If `CopySource` is undefined, this function throws a TypeError. Looks like we should check that `CopySource` is defined before checking its validity.

Reply:
Overlooked this one, will add the check.